crates/turbopack-ecmascript/src/references/mod.rs

```diff
@@ -1062,6 +1064,9 @@ pub(crate) async fn analyze_ecmascript_module(
             let linker = |value| value_visitor(source, origin, value, environment);
             let effects = take(&mut var_graph.effects);
             let link_value = |value| link(&var_graph, value, &linker, &cache);
+            // There can be many references to import.meta, but only the first should hoist
+            // the object allocation.
+            let mut first_import_meta = true;
```

Comment:
I dislike that there is some state during parsing. I feel like this will make the life harder in future. Technically we don't need this info here. We only want to make sure to not emit some declaration multiple times during code genereation. So maybe this should be a code generation state instead.